### PR TITLE
refactor: Add MessageRelay as a preparation for enabling us to migrate responders to ZIO DEV-1728

### PR DESCRIPTION
**Pull request by seakayone**
### Task Description/Number

Issue Number: DEV-1728

For a more detailed explanation take a look at the ADR contained in this PR: https://github.com/dasch-swiss/dsp-api/blob/wip/introduce-MessageRelay/docs/05-internals/design/adr/ADR-0007-zio-fication-of-responders.md

Adds the `MessageRelay` as a central component which is capable of relaying message to subscribed `MessageHandler`s.
Should replace the ask pattern with the `ActorRef` to our `RoutingActor`
 `Responder`s which are migrated to `ZIO`.
 
 This is only meant to be replacing the `RoutingActor` short term.
 In the long run we will prefer to invoke methods on the respective responders/services/repos directly.
 
 However, with the current architecture there are some circular dependencies between the responders and
 the router such that breaking these up is a task for after the ZIO migration.
 
_Out of scope for this PR_

The actual migration of a `Responder`. As a proof of concept that this is working I have started to migrate a responder in another branch based on this PR. If you like to check that take a look at:  https://github.com/dasch-swiss/dsp-api/pull/2456

PR https://github.com/dasch-swiss/dsp-api/pull/2459 contains a complete migration of the responder. I have closed #2456.

### Basic Requirements

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### PR Type

What kind of change does this PR introduce?

- [ ] fix: represents bug fixes
- [x] refactor: represents production code refactoring
- [ ] feat: represents a new feature
- [ ] docs: documentation changes (no production code change)
- [ ] chore: maintenance tasks (no production code change)
- [ ] test: all about tests: adding, refactoring tests (no production code change)
- [ ] other... Please describe:

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] Maybe (not 100% sure => check with FE)

### Does this PR change client-test-data?

- [ ] Yes (don't forget to update the JS-LIB team about the change)
- [x] No

### CODE REVIEW

**Review by BalduinLandolt — APPROVED** [review]
I like it

File: docs/05-internals/design/adr/ADR-0007-zio-fication-of-responders.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+## Status
+
+Draft
+
+## Context
+
+In order to remove all _Akka_ dependencies, we have to migrate the existing `Responder`s to a `ZIO` based
+implementation.
+This migration should be possible to do on a per `Responder` basis so that we do not a single "big-bang" release with
```
Comment (subotic) [inline docs/05-internals/design/adr/ADR-0007-zio-fication-of-responders.md:13]: `so that we do not a` -> `so that we do not do a`

**Review by subotic — APPROVED** [review]

File: webapi/src/main/scala/org/knora/webapi/core/MessageRelay.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+   */
+  def isResponsibleFor(message: ResponderRequest): Boolean
+}
+
+case class MessageRelayLive(handlersRef: Ref[List[MessageHandler]]) extends MessageRelay {
+  override def ask[R: Tag](message: ResponderRequest): Task[R] =
+    handlersRef.get.flatMap { handlers =>
+      ZIO
+        .fromOption(handlers.find(_.isResponsibleFor(message)))
+        .orDieWith(_ => new IllegalStateException(s"No handler defined for ${message.getClass}"))
```
Comment (mpro7) [inline webapi/src/main/scala/org/knora/webapi/core/MessageRelay.scala:88]:
```suggestion
        .orDieWith(_ => new IllegalStateException(s"No handler defined for ${message.getClass}."))
```

**Review by mpro7 — APPROVED** [review]
I only have cosmetic suggestions, please have a look below:

File: docs/05-internals/design/adr/ADR-0007-zio-fication-of-responders.md (hunk 1 above)
Comment (seakayone) [reply docs/05-internals/design/adr/ADR-0007-zio-fication-of-responders.md]: => 955527e56707acb6809391219c7311ba1925edbc

**Pull request merged**